lib/force.go

```diff
@@ -67,6 +67,14 @@ type SessionOptions struct {
 	ApiVersion    string
 	Alias         string
 	RefreshMethod RefreshMethod
+	// RefreshFunc can be set to support refreshing of additional session/refresh types than
+	// available in RefreshMethod. It should return an error on failure.
+	// On success, it is responsible for updating the given Force's credentials.
+	// Usually this is done by calling Force.UpdateCredentials (which persist the changes)
+	// or Force.CopyCredentialAuthFields (which modifies memory only).
+	//
+	// Note that RefreshMethod implementations use Force.UpdateCredentials.
+	RefreshFunc func(*Force) error
```

Comment:
Don't try to serialize `RefreshFunc`:
```suggestion
	RefreshFunc func(*Force) error `json:"-"`
```